pkg/edit/editor_test.go

```diff
@@ -37,6 +37,36 @@ func TestEditor_DoesCommandWithLeadingSpaceToHistory(t *testing.T) {
 	testCommands(t, f.Store /* no commands */)
 }
 
+func TestEditor_TestBeforeAddCmd(t *testing.T) {
+	cases := []struct {
+		name string
+		rc   string
+		in   string
+		exp  []string
+	}{
+		{"nonCallableItem", "edit:add-cmd-filters = [$false]", "echo\n", []string{"echo"}},
+		{"noOutputItem", "edit:add-cmd-filters = [[_]{}]", "echo\n", []string{"echo"}},
+		{"true", "edit:add-cmd-filters = [[_]{ put $true }]", "echo\n", []string{"echo"}},
+		{"false", "edit:add-cmd-filters = [[_]{ put $false }]", "echo\n", nil},
+		{"chain1", "edit:add-cmd-filters = [[_]{ put $false } [_]{ put $true }]", "echo\n", nil},
+		{"chain2", "edit:add-cmd-filters = [[_]{ put $true } [_]{ put $false }]", "echo\n", nil},
+		{"positive", "edit:add-cmd-filters = [[cmd]{ ==s $cmd echo }]", "echo\n", []string{"echo"}},
+		{"negative", "edit:add-cmd-filters = [[cmd]{ ==s $cmd echo }]", "echo x\n", nil},
+	}
```

Comment:
Please add a test case for the short-circuiting behavior. Such a test will need to set up a callback that returns `$false` and another that does some kind of side effect. To check the side effect, the framework of this table-driven test will be insufficient, so write another test.